src/main/scala/com/amazon/deequ/constraints/Constraint.scala

```diff
@@ -916,6 +920,20 @@ object Constraint {
         .getOrElse(0.0)
     }
 
+  private[this] def getUpdatedRowLevelAssertion(assertion: Double => Boolean,
+                                                analyzerOpts: Option[AnalyzerOptions])
+  : java.lang.Double => java.lang.Boolean = {
+    (d: java.lang.Double) => {
+      if (Option(d).isDefined) assertion(d)
+      else analyzerOpts match {
+        case Some(analyzerOptions) => analyzerOptions.filteredRow match {
+          case FilteredRowOutcome.TRUE => true
+          case FilteredRowOutcome.NULL => null
+        }
+        case None => null
```

Comment:
We had discussed that the default `analyzerOptions` behavior should be `FilteredRowOutcome.TRUE`, should we modify 933 to be true? (By default filtered rows are `true` instead of `null`.

Reply:
I was testing out another scenario that may be problematic.
Given the following dataframe:

```
+------+----+----+----+----+-----+-----+
|item  |att1|att2|val1|val2|rule4|rule5|
+------+----+----+----+----+-----+-----+
|1     |a   |f   |1   |1   |true |true |
|22    |b   |d   |2   |NULL|true |true |
|333   |a   |NULL|3   |3   |true |true |
|4444  |a   |f   |4   |4   |true |true |
|55555 |b   |NULL|5   |NULL|true |true |
|666666|a   |f   |6   |6   |true |true |
+------+----+----+----+----+-----+-----+
```

where
```
val analyzerOptions = Option(AnalyzerOptions(filteredRow = FilteredRowOutcome.TRUE))

val min = new Check(CheckLevel.Error, "rule4")
  .hasMin("val2", _ > 0, None, analyzerOptions)
  .where("val1 < 4")
val max = new Check(CheckLevel.Error, "rule5")
  .hasMax("val2", _ < 4, None, analyzerOptions)
  .where("val1 < 4")
```

You'll see that rows 1,2,3 should be skipped -> True
Row 5 should be null as val2 is a null value there.
However, with the above method we convert all nulls to true/null - this doesn't distinguish between null values due to being filtered or null values due to null column values.

Reply:
Thanks @eycho-am for the valuable feedback. The latest PR revision contains a new structure for the column that helps maintain the "source" of a row, whether it is in scope and filtered out. That will help in evaluating the correct outcome for each row.